zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/PermissionsBehavior.java

```diff
@@ -118,4 +119,26 @@ public Either<Rejection, AuthorizationRecord> permissionDoesNotExist(
 
     return Either.right(record);
   }
+
+  public Either<Rejection, AuthorizationRecord> hasValidPermissionTypes(
+      final AuthorizationRecord record) {
+    final var resourceType = record.getResourceType();
+    final var permissionTypes =
+        record.getPermissions().stream()
+            .map(PermissionValue::getPermissionType)
+            .collect(Collectors.toList());
+
+    if (resourceType.getSupportedPermissionTypes().containsAll(permissionTypes)) {
+      return Either.right(record);
+    }
+
+    permissionTypes.removeAll(resourceType.getSupportedPermissionTypes());
+
+    return Either.left(
+        new Rejection(
+            RejectionType.INVALID_ARGUMENT,
+            "Expected to add permission types '%s' for resource type '%s', but these permissions are not supported. Supported permission types are: '%s'"
```

Comment:
:wrench: This method is used on adding and removing so the error message is slightly wrong. I think it's arguable whether we really need to check permission types on removal as you can only remove assigned permissions anyway where we already have the check.

Reply:
Good call, I'll remove it from the removing processor